### DISCUSSION TITLE
Using new wagmi causing ERROR after connecting metamask wallet

### DISCUSSION THREAD

**Discussion by johnpaulcas** [wevm/wagmi#3390] [Q&A]
We're experiencing an issue after connecting to MetaMask using `wagmi version 2.0.3`: we keep getting the error `Uncaught Error: SDK state invalid -- undefined mobile`. Does anyone encounter same issue or know how to fix this? Would really appreciate the help!

**Comment by c0rdeiro** [comment]
I was not able to replicate this, did a simple example and seems to work fine https://stackblitz.com/edit/new-wagmi-aon6zk

**Reply by johnpaulcas** [reply]
The issue occur when adding `metamask()` on connectors in wagmi setup, be able to resolve the issue by removing the metamask as connector, I also notice when i loop through the connectors there  is 2 metamask showing

**Reply by c0rdeiro** [reply]
yeah I altered the connectors to have only `metamask()` but it continues to show 2 connectors in the list, I'm not sure what's causing this

**Comment by c0rdeiro** [comment]
https://github.com/wevm/wagmi/discussions/3068#discussioncomment-8004936 refer to this
